core/controllers/blog_homepage.py

```diff
@@ -167,3 +211,75 @@ def get(self, author_username):
             'summary_dicts': blog_post_summary_dicts
         })
         self.render_json(self.values)
+
+
+class BlogPostSearchHandler(base.BaseHandler):
+    """Provides blog cards for blog search page based on query provided and
+    applied tag filters.
+    """
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS = {}
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {
+            'q': {
+                'schema': {
+                    'type': 'basestring'
+                },
+                'default_value': ''
```

Comment:
Why is this the default value?

Reply:
In case there is no query parameter or tag parameter in the url, default value which should does not influence the search should be used!

Reply:
Why cannot we use `None`?